game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/retreat/Retreater.java

```diff
@@ -0,0 +1,32 @@
+package games.strategy.triplea.delegate.battle.steps.retreat;
+
+import games.strategy.engine.data.Change;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.delegate.battle.MustFightBattle;
+import java.util.Collection;
+import java.util.Map;
+
+interface Retreater {
+
+  enum RetreatLocation {
+    SAME_TERRITORY,
+    OTHER_TERRITORY;
+  }
+
+  Collection<Unit> getRetreatUnits();
+
+  Collection<Territory> getPossibleRetreatSites(Collection<Unit> retreatUnits);
+
+  String getQueryText();
```

Comment:
This seems to be a UI concern, I'm not sure if it belongs in this API as then any implementation is forced to mix UI and logic. Perhaps the caller could do a lookup of query text based on the retreat type?

Reply:
These methods are a "lookup of query text based on the retreat type".  That's what this API is for: handling a specific retreat type with its related change objects and UI text.  Something has to keep the two together.

Reply:
The UI text element seems to make this interface very implementation specific and even coupled to the UI. If the query text is a function of the retreat type, and that is only used in one place, IMO it's a stronger interface if it's made more focused and loses the UI concern. The methods in this interface are generally all related to model, hence it's a bit out of place for the UI parts.

Reply:
I'd prefer having the logic around the different types of retreats together but I've gone ahead and removed the text strings methods.  The caller now has switch statements to figure out which text to use.

Reply:
I'm getting into over-engineering territory, IMO that implies just having multiple interfaces.
`Retreater` & `RetreaterUiLabels`

Implementations can then implement both interfaces, the interfaces themselves would be cohesive/whole, and you can pass around the right interface type depending on your need, if you need the UI portion or if you need the business logic portion. I recognize that we probably have the two concerns mixed in well enough though that we'd either see one interface inherit from the other or both interface types being passed around, both of which would be somewhat ugly.

I appreciate you making that update though, thank you.